build/tools/ExtractRuleDescFromJson/ExtractRuleDescFromJson.csproj

```diff
@@ -13,4 +13,12 @@
     <SonarQubeExclude>true</SonarQubeExclude>
   </PropertyGroup>
 
+  <!-- We're linking to the real SLVS IRuleInfo file to make sure this generator
+       code can't get out of step with the real product code
+  -->
+  <ItemGroup Label="SLVS Linked product code">
+    <Folder Include="SLVS_LinkedProductCode\" />
+    <Compile Include="..\..\..\src\Rules\IRuleInfo.cs" Link="SLVS_LinkedProductCode\IRuleHelp.cs" />
+  </ItemGroup>
```

Comment:
This change is in two parts:
1. changes to the generator tool, so it creates a json file rather than a simple text file, and
2. the corresponding changes in the product code to load the json file

This is not a breaking change; I haven't changed what is returned by the `IRuleMetadataProvider` interface yet.